DataFormats/TrackerCommon/interface/TrackerTopology.h

```diff
@@ -576,7 +577,33 @@ class TrackerTopology {
   std::string print(DetId detid) const;
 
   SiStripDetId::ModuleGeometry moduleGeometry(const DetId &id) const; 
+  
+  // Represents a bitmask that is only valid in some subdetector. 
+  // However, checking validity still has to be done by the caller.
+  struct BitMask {
+    uint32_t startBit_;
+    uint32_t mask_;
+    int32_t  subdet;
+    uint32_t apply(DetId id) const { return ((id.rawId()>>startBit_) & mask_); };
+    bool     valid(DetId id) const { return id.subdetId() == subdet; };
+  };
+
+  std::map<std::string, BitMask> namedPartitions() const { 
+    std::map<std::string, BitMask> namedPartitions_{
```

Comment:
@slava77 thanks for looking at it. Regarding the constructor, things get more delicate with the initialization order then, and moving it to a default-initializer in the .h caused sth. that looks very much like a compiler bug. Also, it is returned by value anyways, so copying vs. in-place construction should not make a big difference. 

Regarding `pbVals() const { return pbVals_; }`: sounds like a good option to me, esp. given that some names (Shell, HalfCylinder) are now added later ayways, since TrackerTopology does not know about them (which raises the question if it should). So, if people are happy with exposing this as an interface, I'd go for it.